app/src/test/contrib/actions/CoverageActionsTests.ts

```diff
@@ -0,0 +1,80 @@
+import { expect } from "chai";
+
+import { Sandbox } from "../../Sandbox";
+import { coverageActionCreators } from "../../../main/contrib/actions/coverageActionCreators";
+import { CoverageService } from "../../../main/contrib/services/CoverageService";
+import {Coverage, CoverageTypes} from "../../../main/contrib/actionTypes/CoverageTypes";
+import {createMockStore} from "../../mocks/mockStore";
+import {mockCoverageSet, mockDemographicDataset, mockScenario, mockTouchstone} from "../../mocks/mockModels";
+import {ScenarioAndCoverageSets} from "../../../main/shared/models/Generated";
+import {statePropsMapHelper} from "../../../main/contrib/helpers/statePropsMapHelper";
+
+describe("Coverage actions tests", () => {
+    const sandbox = new Sandbox();
+
+    const testCoverageSet = mockCoverageSet();
+    const testScenario = mockScenario();
+    const testScenarioAndCoverageSet: ScenarioAndCoverageSets = {
+        coverage_sets: [testCoverageSet],
+        scenario: testScenario
+    };
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it("data sets fetched", (done) => {
+        const store = createMockStore({});
+        sandbox.setStubFunc(statePropsMapHelper, "getResponsibilityIds", ()=>{
+            return Promise.resolve({groupId: "g1", touchstoneId: "t1", scenarioId: "s1"});
+        });
+        sandbox.setStubFunc(CoverageService.prototype, "getDataSets", ()=>{
+            return Promise.resolve(testScenarioAndCoverageSet);
+        });
+
+        store.dispatch(coverageActionCreators.getDataSets())
+        setTimeout(() => {
+            const actions = store.getActions();
+            const expectedPayload = { type: CoverageTypes.COVERAGE_DATA_SETS_FETCHED, data: [testCoverageSet] };
+            expect(actions).to.eql([expectedPayload]);
+            done();
```

Comment:
So previously we had the `checkAsync` helper for this kind of check. I strongly recommend we continue to use this throughout, as it handles the case where an exception is thrown by calling `done(exception)`. Otherwise, if this assertion fails, the test will hang for several seconds and then timeout without us getting proper feedback. Unless mocha has upgraded this behaviour?

Reply:
not , necessary, doesn't hang now

Reply:
Great :)